cmd/schedule-builder/cmd/markdown.go

```diff
@@ -207,6 +225,46 @@ func updatePatchSchedule(refTime time.Time, schedule PatchSchedule, filePath str
 		}
 	}
 
+	newUpcomingReleases := []*PatchRelease{}
+	latestDate := refTime
+	for _, upcomingRelease := range schedule.UpcomingReleases {
+		upcomingTargetDate, err := time.Parse(refDate, upcomingRelease.TargetDate)
+		if err != nil {
+			return fmt.Errorf("parse upcoming release target date: %w", err)
+		}
+
+		if refTime.After(upcomingTargetDate) {
+			logrus.Infof("Skipping outdated upcoming release for %s (%s)", upcomingRelease.Release, upcomingRelease.TargetDate)
+			continue
+		}
+
+		logrus.Infof("Using existing upcoming release for %s (%s)", upcomingRelease.Release, upcomingRelease.TargetDate)
+		newUpcomingReleases = append(newUpcomingReleases, upcomingRelease)
+		latestDate = upcomingTargetDate
+	}
+	for {
+		if len(newUpcomingReleases) >= 3 {
+			logrus.Infof("Got 3 new upcoming releases, not adding any more")
+			break
+		}
+
+		latestDate = latestDate.AddDate(0, 1, 0)
+		cherryPickDay := firstFriday(latestDate)
+		targetDateDay := secondTuesday(latestDate)
+		nextCherryPickDeadline := time.Date(latestDate.Year(), latestDate.Month(), cherryPickDay, 0, 0, 0, 0, time.UTC)
+		nextTargetDate := time.Date(latestDate.Year(), latestDate.Month(), targetDateDay, 0, 0, 0, 0, time.UTC)
+
+		releaseName := nextTargetDate.Format("January 2006")
+		logrus.Infof("Adding new upcoming release for %s", releaseName)
+
+		newUpcomingReleases = append(newUpcomingReleases, &PatchRelease{
+			Release:            releaseName,
+			CherryPickDeadline: nextCherryPickDeadline.Format(refDate),
+			TargetDate:         nextTargetDate.Format(refDate),
+		})
+	}
```

Comment:
I have kind of mixed feelings about this. This will work in most of cases, but it will likely not work around minor releases, big events such KubeCons, and holidays. I know that we can always crosscheck it before merging the update PR, but maybe it's better to have upcoming dates only explicitly defined.

Reply:
We should always cross check the proposed dates and manually adjust them directly in the PR.